src/NuGet.Core/NuGet.Commands/RestoreCommand/Logging/TransitiveNoWarnUtils.cs

```diff
@@ -0,0 +1,527 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.Common;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.ProjectModel;
+using NuGet.Shared;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.DependencyResolver;
+
+namespace NuGet.Commands
+{
+    public class TransitiveNoWarnUtils
```

Comment:
Could this be a static class? It seems like the only reason this isn't currently is because of the cache, but if there were a method that took all graphs at once it could just pass in a cache to the rest of the methods using it.